File: app/src/main/java/com/cv4j/app/fragment/PaintFragment.java
package com.cv4j.app.fragment;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.cv4j.app.R;
import com.cv4j.app.app.BaseFragment;
import com.cv4j.core.filters.OilPaintFilter;
import com.cv4j.core.filters.StrokeAreaFilter;
import com.cv4j.rxjava.RxImageData;
import com.safframework.injectview.Injector;
import com.safframework.injectview.annotations.InjectView;

/**
 * Created by Tony Shen on 2017/5/7.
 */

public class PaintFragment extends BaseFragment {

    @InjectView(R.id.image)
    ImageView image;

    private final static String PAINT_TYPE = "type";

    private final static int DEFAULT_TYPE = 0;
    private final static int OIL_PAINT_TYPE = 1;
    private final static int PENCIL_PAINT_TYPE = 2;

    private int mType;

    public static PaintFragment newInstance(int type) {
        Bundle args = new Bundle();
        args.putInt(PAINT_TYPE, type);
        PaintFragment fragment = new PaintFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle bundle = getArguments();
        if (bundle != null) {
            mType = bundle.getInt(PAINT_TYPE);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_paint, container, false);
        Injector.injectInto(this, v);

        initData();
        return v;
    }

    private void initData() {
        Resources res= getResources();
        Bitmap bitmap = BitmapFactory.decodeResource(res, R.drawable.test_oil_paint);

        switch(mType) {

            case OIL_PAINT_TYPE:
                RxImageData.bitmap(bitmap).addFilter(new OilPaintFilter()).into(image);
                break;

            case PENCIL_PAINT_TYPE:
                RxImageData.bitmap(bitmap).addFilter(new StrokeAreaFilter()).into(image);
                break;

            default:
                image.setImageBitmap(bitmap);
                break;
        }
    }
}

File: app/src/main/java/com/cv4j/app/fragment/PaintFragment.kt
package com.cv4j.app.fragment

import android.content.res.Resources
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.cv4j.app.R
import com.cv4j.app.app.BaseFragment
import com.cv4j.core.filters.OilPaintFilter
import com.cv4j.core.filters.StrokeAreaFilter
import com.cv4j.rxjava.RxImageData.Companion.bitmap
import kotlinx.android.synthetic.main.fragment_paint.*

/**
 *
 * @FileName:
 *          com.cv4j.app.fragment.PaintFragment
 * @author: Tony Shen
 * @date: 2020-05-04 12:54
 * @version: V1.0 <描述当前版本功能>
 */

class PaintFragment : BaseFragment() {

    private var mType = 0

   override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val bundle: Bundle? = arguments
        if (bundle != null) {
            mType = bundle.getInt(PAINT_TYPE)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val v = inflater.inflate(R.layout.fragment_paint, container, false)
        initData()
        return v
    }

    private fun initData() {
        val res: Resources = getResources()
        val bitmap = BitmapFactory.decodeResource(res, R.drawable.test_oil_paint)
        when (mType) {
            OIL_PAINT_TYPE -> bitmap(bitmap).addFilter(OilPaintFilter()).into(image)
            PENCIL_PAINT_TYPE -> bitmap(bitmap).addFilter(StrokeAreaFilter()).into(image)
            else -> image.setImageBitmap(bitmap)
        }
    }

    companion object {
        private const val PAINT_TYPE = "type"
        private const val OIL_PAINT_TYPE = 1
        private const val PENCIL_PAINT_TYPE = 2

        fun newInstance(type: Int): PaintFragment {
            val args = Bundle()
            args.putInt(PAINT_TYPE, type)
            val fragment = PaintFragment()
            fragment.arguments = args
            return fragment
        }
    }
}